test_validate_ceph_config_values_in_rook_config_override failing

The test case given below is failing due to additional config value rbd_mirror_die_after_seconds = 3600. ODF version is 4.10.

Test result - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/16195/testReport/tests.manage.z_cluster.test_ceph_default_values_check/TestCephDefaultValuesCheck/test_validate_ceph_config_values_in_rook_config_override/

This is the same issue reported in #6200. The issue was fixed for ODF version 4.11 only.
```
tests.manage.z_cluster.test_ceph_default_values_check.TestCephDefaultValuesCheck.test_validate_ceph_config_values_in_rook_config_override
 Error Details
AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values are:
  ['', '[global]', 'bdev_flock_retry = 20', 'mon_osd_full_ratio = .85', 'mon_osd_backfillfull_ratio = .8', 'mon_osd_nearfull_ratio = .75', 'mon_max_pg_per_osd = 600', 'mon_pg_warn_max_object_skew = 0', 'mon_data_avail_warn = 15', '[osd]', 'osd_memory_target_cgroup_limit_ratio = 0.8', '']
  The cluster's Ceph values are:['', '[global]', 'rbd_mirror_die_after_seconds = 3600', 'bdev_flock_retry = 20', 'mon_osd_full_ratio = .85', 'mon_osd_backfillfull_ratio = .8', 'mon_osd_nearfull_ratio = .75', 'mon_max_pg_per_osd = 600', 'mon_pg_warn_max_object_skew = 0', 'mon_data_avail_warn = 15', '[osd]', 'osd_memory_target_cgroup_limit_ratio = 0.8', '']
assert Counter({'': ...io = 0.8': 1}) == Counter({'': ...io = 0.8': 1})
  Omitting 11 identical items, use -vv to show
  Left contains 1 more item:
  {'rbd_mirror_die_after_seconds = 3600': 1}
  Full diff:
    Counter({'': 2,
             '[global]': 1,
  +          'rbd_mirror_die_after_seconds = 3600': 1,
             'bdev_flock_retry = 20': 1,
             'mon_osd_full_ratio = .85': 1,
             'mon_osd_backfillfull_ratio = .8': 1,
             'mon_osd_nearfull_ratio = .75': 1,
             'mon_max_pg_per_osd = 600': 1,
             'mon_pg_warn_max_object_skew = 0': 1,
             'mon_data_avail_warn = 15': 1,
             '[osd]': 1,
             'osd_memory_target_cgroup_limit_ratio = 0.8': 1},
    )
```

This issue was encountered once again. Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/7100/296186/296193/296197/log
Run ID: 1670263326
Test Case: test_validate_ceph_config_values_in_rook_config_override
Error Message: AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6976/290813/290820/290824/log
Run ID: 1669952976
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: VSPHERE6 UPI ENCRYPTION 1AZ RHCOS VSAN LSO VMDK 3M 3W tier1 or tier_after_upgrade post upgrade
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6273/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-003vue1cslv33-uba/j-003vue1cslv33-uba_20221201T230828/logs/Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6959/290051/290058/290061/log
Run ID: 1669936593
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-122
OCP Version: 4.12
Job name: RHV IPI 1AZ RHCOS 3M 3W tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6267/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-011ri1c33-t1/j-011ri1c33-t1_20221201T222122/logs/Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6949/289361/289368/289372/log
Run ID: 1669941053
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: VSPHERE6 UPI KMS VAULT V1 1AZ RHCOS VSAN 3M 3W tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6276/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-078vukv11cs33-t1/j-078vukv11cs33-t1_20221201T233022/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6942/289052/289056/289064/log
Run ID: 1669937787
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-122
OCP Version: 4.12
Job name: BAREMETAL UPI 1AZ RHCOS NVME INTEL COMPACT MODE 3M 0W tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6269/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-016bu1cni30-t1/j-016bu1cni30-t1_20221201T223415/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6805/281880/281887/281891/log
Run ID: 1669759366
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: AWS UPI Proxy 3AZ RHCOS 3M 3W tier1 or tier_after_upgrade post upgrade
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6185/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-002aup3c33-uba/j-002aup3c33-uba_20221129T140742/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6794/281166/281173/281177/log
Run ID: 1669738462
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: VSPHERE6 UPI KMS VAULT V1 1AZ RHCOS VSAN 3M 3W tier1 or tier_after_upgrade post upgrade
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6207/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-003vukv11cs33-uba/j-003vukv11cs33-uba_20221129T113107/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6782/280211/280218/280222/log
Run ID: 1669734250
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.13
Job name: AZURE IPI 3AZ RHCOS 3M 3W tier1 or tier_after_upgrade post upgrade
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6204/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-164zi3c33-uo/j-164zi3c33-uo_20221129T092725/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6773/279533/279540/279544/log
Run ID: 1669717720
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: VSPHERE6 UPI ENCRYPTION 1AZ RHCOS VSAN LSO VMDK 3M 3W tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6203/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-092vue1cslv33-t1/j-092vue1cslv33-t1_20221129T092854/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6756/278740/278747/278751/log
Run ID: 1669694775
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: AWS IPI 3AZ RHCOS 3M 3W 3I tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6188/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-212ai3c333-t1/j-212ai3c333-t1_20221129T030853/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6739/277710/277717/277720/log
Run ID: 1669685096
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-120
OCP Version: 4.12
Job name: VSPHERE6 UPI FIPS 1AZ RHCOS VSAN 3M 3W tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6187/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-243vuf1cs33-t1/j-243vuf1cs33-t1_20221129T002117/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

This issue was encountered once again.
Run details:

URL: https://reportportal-ocs4.apps.ocp-c1.prod.psi.redhat.com/ui/#OCS/launches/362/6289/256499/256506/256510/log
Run ID: 1668069131
Test Case: test_validate_ceph_config_values_in_rook_config_override
ODF Build: 4.12.0-91
OCP Version: 4.12
Job name: AWS IPI 3AZ RHCOS 3M 3W tier1
Jenkins job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/6029/
Logs URL: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-060ai3c33-t1/j-060ai3c33-t1_20221110T073408/logs/
Error Message: ```AssertionError: The Ceph config, set by rook-config-override is different than the expected. Please inform OCS-QE about this discrepancy. The expected values ar```

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -266,8 +266,8 @@ DEFAULT_VOLUMESNAPSHOTCLASS_RBD_MS = f"{DEFAULT_CLUSTERNAME}-ceph-rbd"
 
 # encoded value of 'admin'
 ADMIN_USER = "admin"
-GB = 1024**3
-GB2KB = 1024**2
+GB = 1024 ** 3
+GB2KB = 1024 ** 2
 GB2MB = 1024
 
 # Reclaim Policy
@@ -906,7 +906,7 @@ SNO_BOOTSTRAP_IGN = "bootstrap-in-place-for-live-iso.ign"
 TERRAFORM_IGNITION_PROVIDER_VERSION = "v2.1.0"
 
 # Minimum storage needed for vSphere Datastore in bytes.
-MIN_STORAGE_FOR_DATASTORE = 1.1 * 1024**4
+MIN_STORAGE_FOR_DATASTORE = 1.1 * 1024 ** 4
 
 # vSphere related constants
 VSPHERE_NODE_USER = "core"
@@ -1428,9 +1428,9 @@ AUTH_CONFIG_DOCS = (
 TP_CONVERSION = {" B/s": 0.000000976562, " KiB/s": 0.000976562, " MiB/s": 1}
 
 BYTES_IN_KB = 1024
-BYTES_IN_MB = 1024**2
-BYTES_IN_GB = 1024**3
-BYTES_IN_TB = 1024**4
+BYTES_IN_MB = 1024 ** 2
+BYTES_IN_GB = 1024 ** 3
+BYTES_IN_TB = 1024 ** 4
 
 
 # LSO
@@ -1523,7 +1523,7 @@ MAX_OSDS = 18
 
 # Minimum cluster requirements in term of node specs
 MIN_NODE_CPU = 16
-MIN_NODE_MEMORY = 64 * 10**9
+MIN_NODE_MEMORY = 64 * 10 ** 9
 
 # aws tags
 AWS_CLOUDFORMATION_TAG = "aws:cloudformation:stack-name"
@@ -1720,6 +1720,20 @@ osd_memory_target_cgroup_limit_ratio = 0.8
 """
 
 ROOK_CEPH_CONFIG_VALUES_411 = """
+[global]
+bdev_flock_retry = 20
+mon_osd_full_ratio = .85
+mon_osd_backfillfull_ratio = .8
+mon_osd_nearfull_ratio = .75
+mon_max_pg_per_osd = 600
+mon_pg_warn_max_object_skew = 0
+mon_data_avail_warn = 15
+rbd_mirror_die_after_seconds = 3600
+[osd]
+osd_memory_target_cgroup_limit_ratio = 0.8
+"""
+
+ROOK_CEPH_CONFIG_VALUES_412 = """
 [global]
 bdev_flock_retry = 20
 mon_osd_full_ratio = .85

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -1742,7 +1742,6 @@ mon_osd_nearfull_ratio = .75
 mon_max_pg_per_osd = 600
 mon_pg_warn_max_object_skew = 0
 mon_data_avail_warn = 15
-rbd_mirror_die_after_seconds = 3600
 [osd]
 osd_memory_target_cgroup_limit_ratio = 0.8
 """

--- a/ocs_ci/ocs/constants.py
+++ b/ocs_ci/ocs/constants.py
@@ -266,8 +266,8 @@ DEFAULT_VOLUMESNAPSHOTCLASS_RBD_MS = f"{DEFAULT_CLUSTERNAME}-ceph-rbd"
 
 # encoded value of 'admin'
 ADMIN_USER = "admin"
-GB = 1024 ** 3
-GB2KB = 1024 ** 2
+GB = 1024**3
+GB2KB = 1024**2
 GB2MB = 1024
 
 # Reclaim Policy
@@ -906,7 +906,7 @@ SNO_BOOTSTRAP_IGN = "bootstrap-in-place-for-live-iso.ign"
 TERRAFORM_IGNITION_PROVIDER_VERSION = "v2.1.0"
 
 # Minimum storage needed for vSphere Datastore in bytes.
-MIN_STORAGE_FOR_DATASTORE = 1.1 * 1024 ** 4
+MIN_STORAGE_FOR_DATASTORE = 1.1 * 1024**4
 
 # vSphere related constants
 VSPHERE_NODE_USER = "core"
@@ -1428,9 +1428,9 @@ AUTH_CONFIG_DOCS = (
 TP_CONVERSION = {" B/s": 0.000000976562, " KiB/s": 0.000976562, " MiB/s": 1}
 
 BYTES_IN_KB = 1024
-BYTES_IN_MB = 1024 ** 2
-BYTES_IN_GB = 1024 ** 3
-BYTES_IN_TB = 1024 ** 4
+BYTES_IN_MB = 1024**2
+BYTES_IN_GB = 1024**3
+BYTES_IN_TB = 1024**4
 
 
 # LSO
@@ -1523,7 +1523,7 @@ MAX_OSDS = 18
 
 # Minimum cluster requirements in term of node specs
 MIN_NODE_CPU = 16
-MIN_NODE_MEMORY = 64 * 10 ** 9
+MIN_NODE_MEMORY = 64 * 10**9
 
 # aws tags
 AWS_CLOUDFORMATION_TAG = "aws:cloudformation:stack-name"

--- a/tests/manage/z_cluster/test_ceph_default_values_check.py
+++ b/tests/manage/z_cluster/test_ceph_default_values_check.py
@@ -122,6 +122,8 @@ class TestCephDefaultValuesCheck(ManageTest):
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_410.split("\n")
         elif ocs_version >= version.VERSION_4_11:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_411.split("\n")
+        elif ocs_version >= version.VERSION_4_12:
+            stored_values = constants.ROOK_CEPH_CONFIG_VALUES_412.split("\n")
         else:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES.split("\n")
         assert collections.Counter(config_data) == collections.Counter(stored_values), (

--- a/tests/manage/z_cluster/test_ceph_default_values_check.py
+++ b/tests/manage/z_cluster/test_ceph_default_values_check.py
@@ -114,18 +114,21 @@ class TestCephDefaultValuesCheck(ManageTest):
             "confiMap, match the ones stored in ocs-ci"
         )
         ocs_version = version.get_semantic_ocs_version_from_config()
+
         if ocs_version == version.VERSION_4_8:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_48.split("\n")
         elif ocs_version == version.VERSION_4_9:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_49.split("\n")
         elif ocs_version == version.VERSION_4_10:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_410.split("\n")
-        elif ocs_version >= version.VERSION_4_11:
+        elif ocs_version == version.VERSION_4_11:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_411.split("\n")
         elif ocs_version >= version.VERSION_4_12:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES_412.split("\n")
         else:
             stored_values = constants.ROOK_CEPH_CONFIG_VALUES.split("\n")
+        log.info(f"OCS version is {ocs_version}")
+        log.info(f"Stored values are {stored_values}")
         assert collections.Counter(config_data) == collections.Counter(stored_values), (
             f"The Ceph config, set by {constants.ROOK_CONFIG_OVERRIDE_CONFIGMAP} "
             f"is different than the expected. Please inform OCS-QE about this discrepancy. "

